engine/src/main/java/org/terasology/entitySystem/entity/EntityRef.java

```diff
@@ -73,11 +74,17 @@
     public abstract boolean isPersistent();
 
     /**
-     * Sets whether this entity should be saved
+     * To simplify things for the {@link StorageManager} the persistent property of entities can't be changed after
+     * creation. The {@link EntityManager#newBuilder()}} methods can be used to construct non persistent entities.
      *
-     * @param persistent
+     * The existence of this private method will ensure that someone who wants to add this setter again will
+     * read the above comment.
      */
-    public abstract void setPersistent(boolean persistent);
+    @SuppressWarnings("unused")
+    @Deprecated
```

Comment:
Just remove this method. Add some javadoc stating that the persistence flag is immutable after entity creation.